File: libs/document-processor-next/src/__tests__/models/increment-property.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { handlerContractBaseSchema } from '../../model/shared/index.js';

export const incrementPropertySchema = withTypeBlueId('IncrementProperty')(
  handlerContractBaseSchema.extend({
    propertyKey: z.string(),
  })
);

export type IncrementProperty = z.infer<typeof incrementPropertySchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/IncrementProperty.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.processor.model.HandlerContract;

@TypeBlueId("IncrementProperty")
public class IncrementProperty extends HandlerContract {

    private String propertyKey;

    public String getPropertyKey() {
        return propertyKey;
    }

    public void setPropertyKey(String propertyKey) {
        this.propertyKey = propertyKey;
    }
}
